### ISSUE TITLE
Define minimum versions of dependencies

### ISSUE THREAD

**Issue by jorisvandenbossche** [paris-saclay-cds/ramp-board#185]
Point of feedback from the workshop: for quite some packages, we require a recent version, but we don't specify this. So people using their installed packages instead of a fresh environment ran into some errors.
(eg click for the dash vs underscore? `sqlachemy` needs `model_class` keyword. `pandas.concat` needs `sort` keyword)

Ideally we should define and test minimum versions.

**Comment by rth** [comment]
> Ideally we should define and test minimum versions.

Or maybe just pin dependencies? This is an application that should ideally be installed in a separate well-controlled environment, not a library that could be installed everywhere.

Otherwise, with current unpinned versions of dependencies (and the number of dependencies), things are likely to break in 6 month or so, unless someone maintains it continuously.

**Comment by jorisvandenbossche** [comment]
Yep, that is actually true. And certainly given that we can train submissions in a different environment (where you want to have the latest), we can properly pin for ramp-board itself.
